.github/workflows/publish-containers.yml

```diff
@@ -0,0 +1,55 @@
+name: Publish containers
+
+# "on":
+#   push:
+#     tags:
+#       - "v[0-9]+.[0-9]+.[0-9]+"
+#       - "v[0-9]+.[0-9]+.[0-9]+-pre[0-9]+"
+
+on:
+  pull_request:
+jobs:
+  master:
+    strategy:
+      matrix:
+        version: ['0.0.1']
+        build-platform: ['linux/amd64']
+        app: ['explorer']
+    name: Build the ${{ inputs.image_name }} image
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - name: Set up QEMU
+        id: quemu
+        uses: docker/setup-qemu-action@v2
+
+      - name: Setup Docker Buildx
+        uses: docker/setup-buildx-action@v2
+
+      # TODO take version from tag
+      - name: Collect images data
+        id: tags
+        run: |
+          echo '::set-output name=current::vegaprotocol/${{ matrix.app }}:v${{ matrix.version }}'
+          echo '::set-output name=latest::vegaprotocol/${{ matrix.app }}:latest'
+
+      - name: Echo tags
+        run: |
+          echo ${{ steps.tags.outputs.latest }}
+          echo ${{ steps.tags.outputs.current }}
+
+      - name: Login to DockerHub
+        # if: github.event_name != 'pull_request'
+        uses: docker/login-action@v2
+        with:
+          username: ${{ secrets.DOCKERHUB_USERNAME }}
+          password: ${{ secrets.DOCKERHUB_TOKEN }}
+
+      - name: Docker build
+        uses: docker/build-push-action@v3
+        with:
+          context: '.'
+          file: './Dockerfile'
+          build-args: 'APP=${{ matrix.app }}'
+          platforms: ${{ matrix.build-platform }}
+          tags: ${{ steps.tags.outputs.latest }},${{ steps.tags.outputs.current }}
```

Comment:
you need to add `push: true` flag according to the docks -> https://github.com/marketplace/actions/build-and-push-docker-images

Reply:
But yeah, it will be pushing a new version with the same tag because this pipeline means to run every PR a3d version is hardcoded.

Reply:
yeah this just a draft, I wanted a PoC I will change it so it's all reasonable once I have it working I will make it robust. just wanted to get over the initial hurdle of getting it up 👍